BotNet/Bot/UpdateHandler.cs

```diff
@@ -34,14 +36,30 @@ InlineQueryHandler inlineQueryHandler
 			return _me;
 		}
 
-		public async Task HandleUpdateAsync(ITelegramBotClient botClient, Update update, CancellationToken cancellationToken) {
+		public async Task HandleUpdateAsync(ITelegramBotClient botClient, Update update,
+			CancellationToken cancellationToken) {
 			try {
 				switch (update.Type) {
 					case UpdateType.Message:
 						// _logger.LogInformation("Received message from [{firstName} {lastName}]: '{message}' in chat {chatName}.", update.Message!.From!.FirstName, update.Message.From.LastName, update.Message.Text, update.Message.Chat.Title ?? update.Message.Chat.Id.ToString());
 
 						// Retrieve bot identity
 						_me ??= await GetMeAsync(botClient, cancellationToken);
+						
+						// Handle Social Link (better embed) using SocialLinkEmbedFixer 
+						if (update.Message != null) {
+							string message = update.Message.Text ?? update.Message.Caption ?? "";
+							IEnumerable<Uri> possibleUrls = SocialLinkEmbedFixer.GetPossibleUrls(message);
+	
+							foreach (Uri u in possibleUrls.Select(SocialLinkEmbedFixer.Fix))
+							{
+								await botClient.SendTextMessageAsync(
+									chatId: update.Message.Chat.Id,
+									text: u.OriginalString,
```

Comment:
```suggestion
									text: $"Preview: {u.OriginalString}",
```

Biar user tahu ini apaan